src/Web/Static/documents/development/wips/5539-modularize-burn-searches.html.md

```diff
@@ -0,0 +1,152 @@
+        __in BURN_EXTENSION_ENGINE_MESSAGE message,
+        __in const LPVOID pvArgs,
+        __inout LPVOID pvResults,
+        __in_opt LPVOID pvContext
+        );
+
+    enum BURN_EXTENSION_ENGINE_MESSAGE
+    {
+        BURN_EXTENSION_ENGINE_MESSAGE_GETVARIABLENUMERIC,
+        BURN_EXTENSION_ENGINE_MESSAGE_GETVARIABLESTRING,
+        BURN_EXTENSION_ENGINE_MESSAGE_GETVARIABLEVERSION,
+        BURN_EXTENSION_ENGINE_MESSAGE_FORMATSTRING,
+        BURN_EXTENSION_ENGINE_MESSAGE_ESCAPESTRING,
+        BURN_EXTENSION_ENGINE_MESSAGE_EVALUATECONDITION,
+        BURN_EXTENSION_ENGINE_MESSAGE_LOG,
+        BURN_EXTENSION_ENGINE_MESSAGE_SETVARIABLENUMERIC,
+        BURN_EXTENSION_ENGINE_MESSAGE_SETVARIABLELITERALSTRING,
+        BURN_EXTENSION_ENGINE_MESSAGE_SETVARIABLESTRING,
+        BURN_EXTENSION_ENGINE_MESSAGE_SETVARIABLEVERSION,
+    };
+
+2) Create a new `BurnExtension` element for the Burn manifest xml.
+
+    <xs:element name="BurnExtension">
+        <xs:complexType>
+            <xs:attribute name="Id" type="xs:string" />
+            <xs:attribute name="EntryPayloadId" type="xs:string" />
+            <xs:choice minOccurs="0" maxOccurs="unbounded">
+                <xs:element ref="PayloadRef" />
+            </xs:choice>
+        </xs:complexType>
+    </xs:element>
+
+3) Create a new `ExtensionSearch` element for the Burn manifest xml.
+
+    <xs:element name="ExtensionSearch">
+        <xs:annotation>
+            <xs:documentation>
+                The inner XML will contain the information needed for that specific search.
+            </xs:documentation>
+        </xs:annotation>
+        <xs:complexType>
+            <xs:attribute name="Id" type="xs:string" />
+            <xs:attribute name="Variable" type="xs:string" />
+            <xs:attribute name="Condition" type="xs:string" />
+            <xs:attribute name="ExtensionId" type="xs:string" />
+        </xs:complexType>
+    </xs:element>
+
+4) Move the `Util` extension's searches out of the Burn engine and into this new `ExtensionSearch` framework.
+
+5) Move the `WixFileSearch`, `WixRegistrySearch`, `WixComponentSearch`, and `WixProductSearch` tables out of Wix's tables.xml and into the `UtilExtension`'s tables.xml.
+
+6) In `CreateBurnManifestCommand`, it needs to somehow involve the extension in creating the xml.
+Ideally the extension would somehow register a table as a Burn search table, and be able to provide its own subclass of `Row` for each of the rows.
+This class would inherit from a new interface (`ISearchRow`?) which would have a method that returned an instance of `WixSearchInfo`.
+
+However, this ability for an extension to provide its own subclass of `Row` does not exist yet (this was first brought up [here](https://github.com/wixtoolset/wix4/pull/85)). So the alternative would be to create a new Wix extension interface like `IBurnBinderExtension` and get `CreateBurnManifestCommand` to call each of those extensions and generate the xml that way.
+
```

Comment:
`IBurnBackendExtension` exists but there's no manifest extensibility yet.

Reply:
Oh, didn't realize that was created. I guess I need to figure out what Rob's been doing in those micro repos.

Reply:
You'll recognize chunks but lots of new stuff...